src/basic/env-util.c

```diff
@@ -370,9 +370,8 @@ int strv_env_replace(char ***l, char *p) {
 
         assert(p);
 
-        /* Replace first occurrence of the env var or add a new one in the string list. Drop other occurrences. Edits
-         * in-place. Does not copy p.  p must be a valid key=value assignment.
-         */
+        /* Replace first occurrence of the env var or add a new one in the string list. Drop other
+         * occurrences. Edits in-place. Does not copy p. p must be a valid key=value assignment. */
```

Comment:
the fact that `p` is donated is weird... maybe we should rename the function strv_env_replace_donate() or strv_env_replace_consume()

or, at the least, insert a comment right into the argument list `/* donated! */` or so

Reply:
Right, that's a good point. I added a commit to rework strv_env_replace() into strv_env_replace_consume(). It makes almost all callers quite a bit simpler.